src/sentry/tasks/integrations/github/open_pr_comment.py

```diff
@@ -158,34 +159,44 @@ def safe_for_comment(
                 tags={"type": GithubAPIErrorType.UNKNOWN.value, "code": e.code},
             )
             logger.exception("github.open_pr_comment.unknown_api_error", extra={"error": str(e)})
-        return False
+        return False, []
 
     safe_to_comment = True
-    if pullrequest_resp["state"] != "open":
-        metrics.incr(
-            OPEN_PR_METRICS_BASE.format(key="rejected_comment"), tags={"reason": "incorrect_state"}
-        )
-        safe_to_comment = False
-    if pullrequest_resp["changed_files"] > OPEN_PR_MAX_FILES_CHANGED:
-        metrics.incr(
-            OPEN_PR_METRICS_BASE.format(key="rejected_comment"), tags={"reason": "too_many_files"}
-        )
-        safe_to_comment = False
-    if pullrequest_resp["additions"] + pullrequest_resp["deletions"] > OPEN_PR_MAX_LINES_CHANGED:
-        metrics.incr(
-            OPEN_PR_METRICS_BASE.format(key="rejected_comment"), tags={"reason": "too_many_lines"}
-        )
-        safe_to_comment = False
-    return safe_to_comment
 
+    changed_file_count = 0
+    changed_lines_count = 0
+    filtered_pr_files = []
 
-def get_pr_filenames(
-    gh_client: GitHubAppsClient, repository: Repository, pull_request: PullRequest
-) -> List[str]:
-    pr_files = gh_client.get_pullrequest_files(repo=repository.name, pull_number=pull_request.key)
+    for file in pr_files:
+        filename = file["filename"]
+        # don't count the file if it was added or is not a Python file
+        if file["status"] == "added" or not filename.endswith(".py"):
+            continue
+
+        changed_file_count += 1
+        changed_lines_count += file["changes"]
+        filtered_pr_files.append(file)
+
+        if changed_file_count > OPEN_PR_MAX_FILES_CHANGED:
+            metrics.incr(
+                OPEN_PR_METRICS_BASE.format(key="rejected_comment"),
+                tags={"reason": "too_many_files"},
+            )
+            return False, []
+        if changed_lines_count > OPEN_PR_MAX_LINES_CHANGED:
+            metrics.incr(
+                OPEN_PR_METRICS_BASE.format(key="rejected_comment"),
+                tags={"reason": "too_many_lines"},
+            )
+            return False, []
+
+    return safe_to_comment, filtered_pr_files
```

Comment:
Do we need to return a tuple from this function? Would we ever have a `safe_to_comment` be True if there are no filtered pr files? If any empty list or None is returned, would that be sufficient to denote that it's not safe to comment? 

Or restructuring the question, if there were no filtered pr files, where would we even comment?

Reply:
you are absolutely correct

Reply:
this does not need to be a tuple